tests/unit/models/project-test.js

```diff
@@ -461,6 +461,16 @@ describe('models/project.js', function() {
       var addon = project.findAddonByName('not-an-addon');
       expect(addon).to.equal(undefined, 'not found addon should be undefined');
     });
+
+    it('should not return an addon that is a substring of requested name', function() {
+      var addon = project.findAddonByName('foo-ba');
+      expect(addon).to.equal(undefined, 'foo-ba should not be found');
+    })
+
+    it('should return an addon that is the unqualified name of requested name', function() {
+      var addon = project.findAddonByName('qux/foo');
+      expect(addon.name).to.equal('foo', 'should have found the foo addon');
```

Comment:
Can you also add a test searching for just `bar` with `qux/foo-bar`? (It's the inverse case of this test and should not match.)